### PR TITLE
SCMOD-7512: Adding some important debug logging

### PR DESCRIPTION
**Pull request by rahul-kulkarni73**
https://portal.digitalsafe.net/browse/SCMOD-7512

### CODE REVIEW

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
You need to use parameterized logging or otherwise avoid the cost of the parameter construction.  As it is this change will cause significant overhead even when debug mode is not on, because it will always do all the internal processing you've added before it checks the log level.

See https://logback.qos.ch/manual/architecture.html#parametrized.

> The following two lines will yield the exact same output. However, in case of a disabled logging statement, the second variant will outperform the first variant by a factor of at least 30.
> 
> logger.debug("The new entry is "+entry+".");
> logger.debug("The new entry is {}.", entry);

**Comment by rahul-kulkarni73** [comment]
Great information.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
This will avoid the cost of the string concatenation, but in these cases you've still got a significant cost for the construction of those parameters - you need to either create a wrapper object or use the `.isDebugEnabled()` method.

File: worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/MarkupDocumentEngine.java [hunk 1]
```diff
@@ -206,6 +211,9 @@ private MarkupWorkerResult markupDocument(final Multimap<String, ReferencedData>
 
             // Add the list of fields to the
             workerResult.fieldList = XPathHelper.processDocumentWithXPathExpressions(doc, outputFields);
+            if(LOG.isDebugEnabled())
+                LOG.debug("Document after processDocumentWithXPathExpressions \n {}", new XMLOutputter().outputString
+                    (doc));
```
Comment (dermot-hardy) [inline worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/MarkupDocumentEngine.java]: Odd place the break the line.

**Review by dermot-hardy — APPROVED** [review]

File: worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/MarkupDocumentEngine.java (hunk 1 above)
Comment (rahul-kulkarni73) [reply worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/MarkupDocumentEngine.java]: Intellij :)

**Pull request merged**